src/wheatley.ts

```diff
@@ -180,7 +163,14 @@ export class Wheatley extends EventEmitter {
     }
 
     async setup(auth: wheatley_auth) {
-        this.database = await WheatleyDatabase.create(auth);
+        if (!auth.freestanding) {
+            // TODO handle non-freestanding case where no credentials were
```

Comment:
This comment can now be removed

Reply:
This PR added this comment intentionally, because the documented behavior from the README still isn't implemented.

Alternatively, we could also say that Wheatley *does* require a database outside freestanding mode, always.

Reply:
If you turn off freestanding mode you will be loading components which rely on the database so a database is always required outside freestanding mode.

Reply:
> If you turn off freestanding mode you will be loading components which rely on the database so a database is always required outside freestanding mode.

A broader issue is: how are people actually supposed to contribute anything to Wheatley with the system as it stands? There are hardcoded references to TCCPP IDs everywhere, and nothing is configurable in a way that doesn't involve code changes.

In the long run, this is obviously an issue, and not requiring hosting a database is one step in the right direction. Freestanding mode isn't a real solution because it only involves a small number of components.

Reply:
The point is that this feature is needed in the long run; otherwise the hurdle for people to test features locally is not just rewriting IDs and commenting out components, it's also spinning up an entire database, even for more trivial changes.

That's why the TODO should stay in.

Reply:
> A broader issue is: how are people actually supposed to contribute anything to Wheatley with the system as it stands? There are hardcoded references to TCCPP IDs everywhere, and nothing is configurable in a way that doesn't involve code changes.

Yes. It's a difficult problem.
At the moment the way to do it is is a series of local temporary changes: Replace the constants needed for the component you're working on, do what's needed to get things like channels fetched, then enable the component under freestanding.

It would be easy to allow things like channel fetches to fail silently under freestanding mode, which would eliminate the second stop.